Add make-prep.sh script

dhylands opened the pull request:
This automates a bunch of the steps required to make a base image.

mrstegeman commented on image/make-prep.sh:
```diff
@@ -0,0 +1,325 @@
+  HOME_PI="${ROOT_MOUNTPOINT}/home/pi"
+
+  sudo rm -f "${HOME_PI}"/prepare-base*.sh
+  sudo wget --directory-prefix="${HOME_PI}" "https://raw.githubusercontent.com/mozilla-iot/gateway/master/image/prepare-base.sh"
```
Is it necessary to wget these scripts? Why not just copy them from the gateway repo (that this script lives in)?

mrstegeman commented on image/make-prep.sh:
```diff
@@ -0,0 +1,325 @@
+enable_serial_console() {
+
+  CONFIG=${BOOT_MOUNTPOINT}/config.txt
+  if grep -q "enable_uart=1" ${CONFIG} ; then
```
This logic is backwards. This should be `if ! grep ...`.

mrstegeman commented on image/make-prep.sh:
```diff
@@ -0,0 +1,325 @@
+enable_wifi() {
+  echo "Enabling Wifi"
+  WPA_CONF=${ROOT_MOUNTPOINT}/wpa_supplicant/wpa_supplicant.conf
+  sudo sh -c "cat >> '${CONFIG}'" <<END
```
This should be `WPA_CONF`, not `CONFIG`.

Also, what happens if this specific SSID was somehow already enabled? Would the 2 conflict and cause issues?

mrstegeman commented on image/make-prep.sh:
```diff
@@ -0,0 +1,325 @@
+  HOSTS_FILENAME="${ROOT_MOUNTPOINT}/etc/hosts"
+  HOSTNAME_FILENAME="${ROOT_MOUNTPOINT}/etc/hostname"
+  sudo sh -c "echo '${HOSTNAME}' > '${HOSTNAME_FILENAME}'"
+  sudo sed -i "${HOSTS_FILENAME}" -e "s/raspberrypi/${HOSTNAME}/"
```
Might want to add the `g` flag to the regex, in case you have something like `127.0.0.1 raspberrypi raspberrypi.local`.

mrstegeman commented on image/make-prep.sh:
```diff
@@ -0,0 +1,325 @@
+
+  cleanup
+
+  # Because I always forget, print out the commands to upload to aws
```
This comment doesn't seem to match what is being echoed.

mrstegeman commented on image/make-prep.sh:
```diff
@@ -0,0 +1,325 @@
+  fi
+  BOOT_MOUNTED=1
+
+  # Mount the root parition of the image
```
partition

mrstegeman commented on image/make-prep.sh:
```diff
@@ -0,0 +1,325 @@
+  fi
+  LOOP_MOUNT_CREATED=1
+
+  # Mount the boot parition of the image
```
partition

dhylands commented:
Thanks - I addressed all of the review comments and actually tested all of the features this time.

I also added a --dd option to have it write the final image out to an sdcard, and a --summary option which prints out the contents of the adjusted files.
